src/gnssro/gnssro_bufr2ioda_generic.f90

```diff
@@ -0,0 +1,496 @@
+   call check(nf90_put_att(grpid_metadata, varid_sclf, "units", "1"))
+
+   call check(nf90_def_var(grpid_metadata, "satelliteTransmitterId", NF90_INT, nlocs_dimid, varid_ptid))
+   call check(nf90_def_var_fill(grpid_metadata, varid_ptid, 0, i_missing))
+   call check(nf90_def_var_deflate(grpid_metadata, varid_ptid,       &
+                                    & shuffle=1, deflate=1, deflate_level=deflate_level))
+   call check(nf90_put_att(grpid_metadata, varid_ptid, "longname", "GNSS satellite transmitter identifier (1-32)"))
+   call check(nf90_put_att(grpid_metadata, varid_ptid, "units", "1"))
+
+   call check(nf90_def_var(grpid_metadata, "satelliteIdentifier", NF90_INT, nlocs_dimid, varid_said))
+   call check(nf90_def_var_fill(grpid_metadata, varid_said, 0, i_missing))
+   call check(nf90_def_var_deflate(grpid_metadata, varid_said,       &
+                                    & shuffle=1, deflate=1, deflate_level=deflate_level))
+   call check(nf90_put_att(grpid_metadata, varid_said, "longname", &
+      & "Low Earth Orbit satellite identifier, e.g., COSMIC2=750-755"))
+   call check(nf90_put_att(grpid_metadata, varid_said, "units", "1"))
+
+   call check(nf90_def_var(grpid_metadata, "occulting_sat_is", NF90_INT, nlocs_dimid, varid_siid))
+   call check(nf90_def_var_deflate(grpid_metadata, varid_siid,       &
+                                    & shuffle=1, deflate=1, deflate_level=deflate_level))
+   call check(nf90_put_att(grpid_metadata, varid_siid, "longname", "satellite instrument"))
+   call check(nf90_put_att(grpid_metadata, varid_siid, "units", "1"))
+   call check(nf90_def_var_fill(grpid_metadata, varid_siid, 0, i_missing))
+
+   call check(nf90_def_var(grpid_metadata, "qualityFlags", NF90_INT, nlocs_dimid, varid_qcfg))
+   call check(nf90_def_var_deflate(grpid_metadata, varid_qcfg,       &
+                                    & shuffle=1, deflate=1, deflate_level=deflate_level))
+   call check(nf90_put_att(grpid_metadata, varid_qcfg, "longname", "RO QC flags based on the 16-digit binary table of WMO"))
+
+   call check(nf90_def_var(grpid_metadata, "satelliteAscendingFlag", NF90_INT, nlocs_dimid, varid_asce))
+   call check(nf90_def_var_deflate(grpid_metadata, varid_asce,       &
+                                    & shuffle=1, deflate=1, deflate_level=deflate_level))
+   call check(nf90_put_att(grpid_metadata, varid_asce, "longname", "the original occultation ascending/descending flag"))
+   call check(nf90_put_att(grpid_metadata, varid_asce, "valid_range", int((/0, 1/))))
+   call check(nf90_put_att(grpid_metadata, varid_asce, "flag_values", int((/0, 1/))))
+   call check(nf90_put_att(grpid_metadata, varid_asce, "flag_meanings", "descending ascending"))
+   call check(nf90_put_att(grpid_metadata, varid_asce, "units", "1"))
+   call check(nf90_def_var_fill(grpid_metadata, varid_asce, 0, i_missing))
+
+   call check(nf90_def_var(grpid_metadata, "process_center", NF90_INT, nlocs_dimid, varid_ogce))
```

Comment:
I believe the name should be `dataProviderOrigin` rather than `process_center`.

Reply:
Done! thanks!